src/dev/preflight_check/checks/check_jest.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { dirname } from 'path';
+import execa from 'execa';
+import { findFileUpwards } from '../utils/find_file_upwards';
+import { PreflightCheck, TestResponse } from './preflight_check';
+
+export class JestCheck extends PreflightCheck {
+  id = 'jest';
+
+  public async runCheck() {
+    const files = Array.from(this.files.values());
+    const response: TestResponse = { test: this.id, errors: [] };
+
+    if (files.length === 0) {
+      return response;
+    }
+
+    for (const { path } of files) {
+      const jestConfig = await findFileUpwards(dirname(path), 'jest.config.js');
+
+      if (!jestConfig) {
+        // Could not find jest.config.js for ${path}
+        return response;
+      }
+
+      try {
+        await execa('npx', ['jest', path, '-c', jestConfig], {
```

Comment:
Can we swap the `npx` calls with `node_modules/.bin`?  There was some discussion earlier on wanting deterministic versions, and npx will install latest if it's not already installed.

https://docs.npmjs.com/cli/v10/commands/npx
> If any requested packages are not present in the local project dependencies, then they are installed to a folder in the npm cache,